### PR TITLE
feat(tokens): Add granular component tokens (VIV-1059)

### PR DESCRIPTION
**Pull request by acelletti**
This PR aims to add individual CSS variables that can be overridden for a specific component & given connotation.

This provides a flexible approach that can be used to customise the appearance of the web components for different projects including but not limited to CPaaS.

### CODE REVIEW

**Review by gfrileux — COMMENTED** [review]
Love it ! great start towards achieving CPaaS !

**Review by rachelbt — CHANGES_REQUESTED** [review]
the change apply only to the idle state but not to all the other states such as: over, active, selected/checked:
image

**Review by rachelbt — COMMENTED** [review]
I like it much better the samples outside of the docs :)

Some of the components are underlying so we don't need them in that page (went over some but nat all)
Also I'm not sure about the tabs of the different connotations.

Thought about adding a small explanation at the top - something like this:
https://docs.google.com/document/d/1TLea6p8IuRQ55gSzf_se1KxV-tqrqGKV6qTM94aKF04/edit?usp=sharing
What do you say?

Some points I'de like us to think of:
- Maybe we need to find a better naming for the vars of splitting them into appearances (ghost, filled...).
- Some of the colors that are set are colors for disabled state - are we providing them for change as well?
- some of the colors are via variables inside the components and not through the mixin map - we need to think about those as well.

File: docs/design/customization.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+## Listbox
+
+```html variables-preview[listbox]
+<vwc-listbox>
+  <vwc-option value="1" text="Option" selected></vwc-option>
+  <vwc-option value="2" text="Option"></vwc-option>
+  <vwc-option value="3" text="Option"></vwc-option>
+</vwc-listbox>
+```
+
```
[the remark this answers is not in the document]
Comment (RichardHelm) [reply docs/design/customization.md]: Sure, I've removed it now

File: docs/design/customization.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+```html variables-preview[fab]
+<vwc-fab connotation="$CONNOTATION" icon='cart-line' label='Add to cart'></vwc-fab>
+```
+
+## Icon
+
+```html variables-preview[icon]
+<vwc-icon size="3" name="heart-solid" connotation="$CONNOTATION"></vwc-icon>
+```
```
[the remark this answers is not in the document]
Comment (RichardHelm) [reply docs/design/customization.md]: Removed it as well

File: docs/design/customization.md [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+		<td><code>fierce</code></td>
+	</tr>
+	<tr>
+		<td class="scale"><code>--800</code></td>
+		<td><div class="color-square" style="background-color: var(--vvd-color-cta-800)"></td>
+		<td><code>contrast</code></td>
+	</tr>
+	</tbody>
+</table>
+```
```
Comment (rachelbt) [inline docs/design/customization.md:81]:
```suggestion
<style>
.scale {
	text-align: right;
}

.color-square {
	width: 22px;
	aspect-ratio: 1;
	border: 1px solid var(--vvd-color-neutral-300);
  background-color: var(--sample-bg);
}
</style>

<vwc-layout style="--layout-grid-template-columns: auto 22px 1fr; --layout-column-gap: 8px; --layout-row-gap: 6px;">
		<span class="scale"><code>--var-color-50</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-cta-50)"></span>
		<span><code>faint</code>, <code>backdrop</code></span>
		<span class="scale"><code>--var-100</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-cta-100)"></span>
		<span><code>soft</code></span>
		<span class="scale"><code>--var-200</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-cta-200)"></span>
		<span><code>dim</code></span>
		<span class="scale"><code>--var-300</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-cta-300)"></span>
		<span><code>pale</code></span>
		<span class="scale"><code>--var-400</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-cta-400)"></span>
		<span><code>light</code></span>
		<span class="scale"><code>--var-color-500</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-cta-500)"></span>
		<span><code>primary</code>, <code>intermediate</code></span>
		<span class="scale"><code>--var-color-600</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-cta-600)"></span>
		<span><code>firm</code>, <code>primary-increment</code></span>
		<span class="scale"><code>--var-color-700</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-cta-700)"></span>
		<span><code>fierce</code></span>
		<span class="scale"><code>--var-color-800</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-cta-800)"></span>
		<span><code>contrast</code></span>
		<span class="scale"><code>--var-color-canvas</code></span>
		<span class="color-square" style="--sample-bg: var(--vvd-color-canvas)"></span>
		<span><code>primary-text</code></span>
</vwc-layout>
```

I dont see the point in having this as an HTML preview.
Maybe the css can be at main and not here :)

**Review by rachelbt — COMMENTED** [review]
changed the doc a bit.
Why do we have 2 images of the css-variable structure in 2 places?

**Review by rachelbt — COMMENTED** [review]
changed the doc a bit.
Why do we have 2 images of the css-variable structure in 2 places?

**Review by RichardHelm — APPROVED** [review]

**Pull request merged**